app/products/calls/state/actions.ts

```diff
@@ -0,0 +1,276 @@
+    setCurrentCall(null);
+};
+
+export const callStarted = (serverUrl: string, call: Call) => {
+    const callsState = getCallsState(serverUrl);
+    const nextCalls = {...callsState.calls};
+    nextCalls[call.channelId] = call;
+    setCallsState(serverUrl, {...callsState, calls: nextCalls});
+
+    const nextChannelsWithCalls = {...getChannelsWithCalls(serverUrl), [call.channelId]: true};
+    setChannelsWithCalls(serverUrl, nextChannelsWithCalls);
+};
+
+// TODO: should be called callEnded to match the ws event. Will fix when callEnded is implemented.
+export const callFinished = (serverUrl: string, channelId: string) => {
+    const callsState = getCallsState(serverUrl);
+    const nextCalls = {...callsState.calls};
+    delete nextCalls[channelId];
+    setCallsState(serverUrl, {...callsState, calls: nextCalls});
+
+    const channelsWithCalls = getChannelsWithCalls(serverUrl);
+    const nextChannelsWithCalls = {...channelsWithCalls};
+    delete nextChannelsWithCalls[channelId];
+    setChannelsWithCalls(serverUrl, nextChannelsWithCalls);
+};
+
+export const setUserMuted = (serverUrl: string, channelId: string, userId: string, muted: boolean) => {
+    const callsState = getCallsState(serverUrl);
+    if (!callsState.calls[channelId] || !callsState.calls[channelId].participants[userId]) {
+        return;
+    }
+
+    const nextUser = {...callsState.calls[channelId].participants[userId], muted};
+    const nextCall = {
+        ...callsState.calls[channelId],
+        participants: {...callsState.calls[channelId].participants},
+    };
+    nextCall.participants[userId] = nextUser;
+    const nextCalls = {...callsState.calls};
+    nextCalls[channelId] = nextCall;
+    setCallsState(serverUrl, {...callsState, calls: nextCalls});
+
+    // Was it the current call? If so, update that too.
+    const currentCall = getCurrentCall();
+    if (!currentCall || currentCall.channelId !== channelId) {
+        return;
+    }
+
+    const nextCurrentCall = {
+        ...currentCall,
+        participants: {
+            ...currentCall.participants,
+            [userId]: {...currentCall.participants[userId], muted},
+        },
+    };
+    setCurrentCall(nextCurrentCall);
+};
+
+export const setRaisedHand = (serverUrl: string, channelId: string, userId: string, timestamp: number) => {
+    const callsState = getCallsState(serverUrl);
+    if (!callsState.calls[channelId] || !callsState.calls[channelId].participants[userId]) {
+        return;
+    }
+
+    const nextUser = {...callsState.calls[channelId].participants[userId], raisedHand: timestamp};
+    const nextCall = {
+        ...callsState.calls[channelId],
+        participants: {...callsState.calls[channelId].participants},
+    };
+    nextCall.participants[userId] = nextUser;
+    const nextCalls = {...callsState.calls};
+    nextCalls[channelId] = nextCall;
+    setCallsState(serverUrl, {...callsState, calls: nextCalls});
+
+    // Was it the current call? If so, update that too.
```

Comment:
For my own understanding, can we have concurrent calls ?

Reply:
Here I mean two calls in parallel and the user switching between the two.

Reply:
Nope, not yet.